### PR TITLE
VZ-8067 Export metrics from the Cluster Operator

### PR DESCRIPTION
**Pull request by davisgomes**
Provide a summary of the change and which issue (i.e. ticket) is fixed.
If there are any dependencies, for example on a PR in another repository, list those.

### CODE REVIEW

File: cluster-operator/controllers/rancher/rancher_cluster_controller.go [hunk 1]
```diff
@@ -45,6 +48,21 @@ var gvk = schema.GroupVersionKind{
...
+	})
+	reconcileErrorCount = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "vz_cluster_operator_reconcile_errors_total",
+		Help: "The amount of errors encountered in the reconcile process",
+	})
+	reconcileSuccessCount = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "vz_cluster_operator_reconcile_success_total",
+		Help: "The number of times the reconcile process succeeded",
+	})
+)
```
Comment (fredtibbitts) [inline cluster-operator/controllers/rancher/rancher_cluster_controller.go:64]: This operator has two controllers: this one for Rancher clusters and another one for VMCs. Should we have another similar set of metrics in vmc_controller.go? These names and descriptions would need to reflect they are for the Rancher clusters. Or was that something you planned to do in a follow-on PR?

**Review by fredtibbitts — APPROVED** [review]
Looks like a good start. Do we need to add/change any network policies or Istio auth policies to allow Prometheus scraping? We may have already done that, but if you tested in a cluster with Calico enabled and scraping works then we should be good.

File: cluster-operator/controllers/rancher/rancher_cluster_controller.go [hunk 2]
```diff
@@ -45,6 +48,21 @@ var gvk = schema.GroupVersionKind{
 	Kind:    "Cluster",
 }
 
+var (
+	reconcileTimeMetric = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "vz_cluster_operator_reconcile_cluster_duration_seconds",
+		Help: "The duration of the reconcile process for cluster objects",
+	})
+	reconcileErrorCount = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "vz_cluster_operator_reconcile_errors_total",
```
Comment (knelson7) [inline cluster-operator/controllers/rancher/rancher_cluster_controller.go:57]: Re metric naming conventions as of now, we're using error singular—"vz_cluster_operator_reconcile_error_total". Forgot to mention it when you asked, my apologies.

File: cluster-operator/controllers/rancher/rancher_cluster_controller.go (hunk 1 above)
Comment (davisgomes) [reply cluster-operator/controllers/rancher/rancher_cluster_controller.go:64]: Yes. I can add these for the other reconcile loop.

File: cluster-operator/controllers/rancher/rancher_cluster_controller.go (hunk 2 above)
Comment (davisgomes) [reply cluster-operator/controllers/rancher/rancher_cluster_controller.go:57]: Thanks. I will update these names in the following push

**Review by knelson7 — APPROVED** [review]

**Pull request merged**